samples/python_interop/cirq_samples/teleportation_basics.ipynb

```diff
@@ -0,0 +1,188 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "a0b520fb",
+   "metadata": {},
+   "source": [
+    "# Quantum Teleportation in Cirq\n",
+    "\n",
+    "This notebook walks through constructing a minimal quantum teleportation protocol using basic Cirq operations. We'll explore entanglement, state preparation, measurement, and classical feed-forward corrections."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ade75f5f",
+   "metadata": {},
+   "source": [
+    "## 1. Imports"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a8201f1b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cirq\n",
+    "from typing import Sequence"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "deeaace0",
+   "metadata": {},
+   "source": [
+    "## 2. Helper Functions\n",
+    "We define small helpers for Bell pair preparation and message encoding."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4557aa98",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def PrepareBellPair(q1: cirq.Qid, q2: cirq.Qid) -> Sequence[cirq.Operation]:\n",
+    "    return [\n",
+    "        cirq.H.on(q1),\n",
+    "        cirq.CNOT.on(q1, q2)\n",
+    "    ]\n",
+    "\n",
+    "def PrepareMessage(alice: cirq.Qid, message: cirq.Qid) -> Sequence[cirq.Operation]:\n",
+    "    return [\n",
+    "        cirq.CNOT.on(message, alice),\n",
+    "        cirq.H.on(message)\n",
+    "    ]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b9e379d2",
+   "metadata": {},
+   "source": [
+    "## 3. Building the Teleportation Operation List\n",
+    "We build the protocol step-by-step."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3be71578",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def simple_teleportation() -> Sequence[cirq.Operation]:\n",
+    "    alice = cirq.NamedQubit('alice')\n",
+    "    bob = cirq.NamedQubit('bob')\n",
+    "    program: Sequence[cirq.Operation] = []\n",
+    "    # 1. Create entangled pair shared by Alice and Bob\n",
+    "    program.extend(PrepareBellPair(alice, bob))\n",
+    "    # 2. Prepare a message qubit (state to teleport)\n",
+    "    msg = cirq.NamedQubit('msg')\n",
+    "    program.append(cirq.rx(0.7)(msg))  # Example arbitrary state\n",
+    "    # 3. Entangle message with Alice's half\n",
+    "    program.extend(PrepareMessage(alice, msg))\n",
+    "    # 4. Measure Alice + msg to obtain classical bits\n",
+    "    program.append(cirq.measure(msg, key='b1'))\n",
+    "    program.append(cirq.measure(alice, key='b2'))\n",
+    "    # 5. Classical feed-forward corrections on Bob\n",
```

Comment:
For those who sees teleportation for the first time, maybe it would help to explicitly mark a place where classical data is communicated to the Bob's site.

Reply:
Good idea! Done.